src/plugin.ts

```diff
@@ -158,6 +108,16 @@ export abstract class ExtismPluginBase {
       }
     }
 
+    for (const m in imports) {
+      if (m === "wasi_snapshot_preview1") {
+        continue;
+      }
+
+      for (const f in imports[m]) {
+        imports[m][f] = imports[m][f].bind(this.currentPlugin);
```

Comment:
if you call `.bind(null, this.currentPlugin)` you can move the `CurrentPlugin` into the arglist (vs as a `thisarg`)!

```suggestion
        imports[m][f] = imports[m][f].bind(null, this.currentPlugin);
```

Reply:
ah, good catch. I had confused myself because I basically wanted to make it optional (i.e. only pass it if the first argument is actually a CurrentPlugin). But now that think about it, I guess it's simpler to always pass it in as the first parameter